Feature flags: Removes various production-enabled flags.

cpapazoglou opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to p58i-j6R-p2

## Flags Removed

* calypsoify/plugins
* jetpack/agency-dashboard
* jetpack/plugin-management
* performance-profiler/llm
* performance-profiler/logged-in
* plugins/multisite-scheduled-updates
* plugins/ssr-categories
* plugins/ssr-details
* plugins/ssr-landing

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Visual inspection should be enough.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
Removes various production-enabled feature flags.

epeicher commented on client/jetpack-cloud/sections/sidebar-navigation/manage-selected-site.tsx:
```diff
@@ -190,7 +187,7 @@ const useMenuItems = ( {
 			isAdmin,
 			isAgency,
-			isPluginManagementEnabled,
+			showScanHistory,
```
Thanks for adding the missing dependency! 🙌

epeicher reviewed (approved):
I have gone through all the changes and they LGTM. Thank you very much for improving the codebase

epeicher committed:
Fix site-table test

epeicher committed:
Fix test in useRowMetadata hook

epeicher commented:
Hi @cpapazoglou, I have updated the tests in commits [bb019a9](https://github.com/Automattic/wp-calypso/pull/98344/commits/bb019a9319fd1261288b93553cca409eb80e5898) and [66376c5](https://github.com/Automattic/wp-calypso/pull/98344/commits/66376c5bc26a075c03dfa34e88e620ebc2e0529d) because in the tests the flag `jetpack/plugin-management` was `false` in [this line of code](https://github.com/Automattic/wp-calypso/pull/98344/files#diff-efd478eadeccf7bd83f5b1e6700e7666671e4cf46f8920405a9db072156f4fa6L70) so I have updated the tests to reflect the new behaviour, which is entering the condition.

cpapazoglou commented:
> Hi @cpapazoglou, I have updated the tests in commits [bb019a9](https://github.com/Automattic/wp-calypso/pull/98344/commits/bb019a9319fd1261288b93553cca409eb80e5898) and [66376c5](https://github.com/Automattic/wp-calypso/pull/98344/commits/66376c5bc26a075c03dfa34e88e620ebc2e0529d) because in the tests the flag `jetpack/plugin-management` was `false` in [this line of code](https://github.com/Automattic/wp-calypso/pull/98344/files#diff-efd478eadeccf7bd83f5b1e6700e7666671e4cf46f8920405a9db072156f4fa6L70) so I have updated the tests to reflect the new behaviour, which is entering the condition.

Thanks for fixing the tests, I haven't noticed them 🤦
